src/datamodel/serializer/Uuid.cpp

```diff
@@ -10,3 +10,122 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/serializer/Uuid.h"
+
+#include <arpa/inet.h>
+
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::serializer::uuid {
+
+std::string AsString::serialize(const uprotocol::v1::UUID uuid) {
+	// Extracting the parts of the UUIDv8
+	uint64_t unix_ts_ms = (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;  // 48 bit
+	uint8_t ver = (uuid.msb() >> 12) & 0xF;                     // 4 bit
+	uint16_t counter = uuid.msb() & 0xFFF;                      // 12 bit
+	uint8_t var = (uuid.lsb() >> 62) & 0x3;                     // 2 bit
+	uint64_t rand_b = uuid.lsb() & 0x3FFFFFFFFFFFFFFF;          // 62 bit
+
+	// Formatting the UUIDv8 in the traditional format
+	std::stringstream ss;
+	ss << std::hex << std::setfill('0') << std::setw(8)
+	   << ((unix_ts_ms >> 16) & 0xFFFFFFFF)  // First 32 bits of timestamp
+	   << "-" << std::setw(4)
+	   << ((unix_ts_ms)&0xFFFF)  // Next 16 bits of timestamp i.e. last 16 bit
+	                             // of ts
+	   << "-" << std::setw(4)
+	   << (((ver & 0xF) << 12) |
+	       (counter & 0xFFF))  // Last 16 bits of timestamp and version
+	   << "-" << std::setw(4)
+	   << (((var & 0x3) << 14) |
+	       ((rand_b >> 48) & 0x3FFF))  // Variant and randb
+	   << "-" << std::setw(12) << (rand_b & 0xFFFFFFFFFFFF);  // Random number
+
+	return std::move(ss).str();
+}
+
+uprotocol::v1::UUID AsString::deserialize(const std::string& str) {
+	// Check if the UUID string is in the correct format
+	if (str.length() != 36 || str[8] != '-' || str[13] != '-' ||
+	    str[18] != '-' || str[23] != '-') {
+		throw std::invalid_argument("Invalid UUID string format");
+	}
+
+	uprotocol::v1::UUID uuid;
+	uint64_t unix_ts_ms = 0;
+	uint8_t ver = 0;
+	uint16_t counter = 0;
+	uint8_t var = 0;
+	uint64_t rand_b = 0;
+
+	try {
+		// Extract the parts from the UUID string
+		unix_ts_ms = std::stoull(str.substr(0, 8), nullptr, 16) << 16;
+		unix_ts_ms |= std::stoull(str.substr(9, 4), nullptr, 16);
+
+		uint16_t ver_counter = std::stoul(str.substr(14, 4), nullptr, 16);
+		ver = (ver_counter >> 12) & 0xF;
+		counter = ver_counter & 0xFFF;
+
+		uint16_t var_randb = std::stoul(str.substr(19, 4), nullptr, 16);
+		var = (var_randb >> 14) & 0x3;
+		rand_b = static_cast<uint64_t>(var_randb & 0x3FFF) << 48;
+		rand_b |= std::stoull(str.substr(24), nullptr, 16);
+	} catch (const std::exception& e) {
+		throw std::invalid_argument("Invalid UUID string format");
+	}
```

Comment:
NITPICK: Sorry :(  -- again maybe a comment showing how the UUID is deserialized

// format : 12345678-1234-1234-1234-1234567890123
// layout : __msb___-_msb-vcnt-____-_____RAND___
// v - version
// cnt -counter

OR whatever - I know you know this stuff now, but in 6 months we're ALL going to forget it.